src/rust/engine/src/context.rs

```diff
@@ -179,29 +181,46 @@ impl Core {
       exec_strategy_opts.local_cleanup,
     );
 
-    let maybe_nailgunnable_local_command_runner: Box<dyn CommandRunner> =
+    let maybe_nailgunnable_local_command_runner: Result<Box<dyn CommandRunner>, String> =
       if exec_strategy_opts.local_enable_nailgun {
-        Box::new(nailgun::CommandRunner::new(
-          local_command_runner,
-          local_execution_root_dir.to_path_buf(),
-          store.clone(),
-          executor.clone(),
-          // We set the nailgun pool size to twice the number that will ever be active in order to
-          // keep warm but idle processes for task fingerprints which are not currently busy (e.g.
-          // while busy running scalac, keeping some javac processes idle).
-          // TODO: The nailgun pool size should be configurable independent of concurrency, along
-          // with per-instance memory usage. See https://github.com/pantsbuild/pants/issues/13067.
-          exec_strategy_opts.local_parallelism * 2,
-        ))
+        if exec_strategy_opts.child_default_memory > exec_strategy_opts.child_max_memory {
+          Err(format!(
+            "\
```

Comment:
It would be a little bit better to do this validation here: https://github.com/pantsbuild/pants/blob/c1a6dd1ad90b255c00defeb1e85e35e5fe90a504/src/python/pants/option/global_options.py#L1534-L1544

...then here you could do `max(1, exec_strategy_opts.child_max_memory / exec_strategy_opts.child_default_memory)` rather than validating.

Reply:
yeah, that makes total sense, much better than validating in the Rust side as that was making pantsd panic.